### PR TITLE
Add basic security policy

### PR DESCRIPTION
**Pull request by agjohnson**
This structure and content was inspired by Help Scout's security policy
doc: https://www.helpscout.com/company/legal/security/

### CODE REVIEW

File: docs/legal/security-policy.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    All documentation, application dashboard, and API access is transmitted
+    using SSL encryption. We do not support unencrypted requests, even for
+    public project documentation hosting.
+
+Temporary repository storage
+    We do not store or cache user repository data, temporary storage is used for
+    every project build on Read the Docs.
+
+Authentication
+    Read the Docs supports SSO GitHub, GitLab, Bitbucket and Google Workspaces
```
Comment (ericholscher) [inline docs/legal/security-policy.rst:53]:
```suggestion
    Read the Docs supports SSO with GitHub, GitLab, Bitbucket, and Google Workspaces
```

**Review by ericholscher — APPROVED** [review]
Good summary. 👍

**Comment by agjohnson** [comment]
Also, I cooked this up very quickly. There might be some missing items we might want in this document

**Review by davidfischer — APPROVED** [review]
I understand this has a different goal than https://docs.readthedocs.io/en/stable/security.html but I think they should cross-link.

**Comment by agjohnson** [comment]
> I understand this has a different goal than https://docs.readthedocs.io/en/stable/security.html but I think they should cross-link.

Makes sense. We can come back on this point perhaps, I have another request for this document

**Pull request merged**